docs/guide/config/toolbar-colors.md

```diff
@@ -35,29 +35,91 @@ in in your toolbar configuration.
 
-An example `palette.gpl` file is shown below ([source](https://github.com/xournalpp/xournalpp/issues/1812#issuecomment-723679463)):
+Prerequisite: Xournal++ ran at least once with the new version
+
+ 1. Open `palette.gpl` in the [config file location](../file-locations.md)
+ 2. Edit the `palette.gpl` to your needs
+ 3. Restart Xournal++
+
+### .gpl File Format
+
+!!! note
+
+    The `.gpl` file format is not authoritatively defined anywhere.
+    Hence the below definition might only be valid for Xournal++.
+    However, we are trying to be compatible with `.gpl` files as used in other Software.
+    In case you find a case where the `.gpl` file is not parsable by Xournal++ but you think it should be parsable since for example GIMP accepts it, please file a bug report.
+
+Lets start with an example:
 
 ```
 GIMP Palette
-Name: Xournal New Palette.gpl
+Name: Xournal Default Palette
 #
-255 225 107 #FFE16B
-255 161  84 #FFA154
-205 158 247 #CD9EF7
-155 219  77 #9BDB4D
-100 186 255 #64BAFF
-128 128 128 #808080
- 58 145   4 #3A9104
-237  83  83 #ED5353
-  0  46 153 #002E99
-  0   0   0 #000000
-255 255 255 #FFFFFF
+0 0 0 Black
+0 128 0 Green
+0 192 255 Light Blue
+0 255 0 Light Green
+51 51 204 Blue
+128 128 128 Gray
+255 0 0 Red
+255 0 255 Magenta
+255 128 0 Orange
+255 255 0 Yellow
+255 255 255 White
 ```
 
-[mdn color picker]: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Colors/Color_picker_tool
+This is the default color palette of Xournal++, which is created if no palette is found.
+
+It consists of several Parts:
+
+ 1. Header: `GIMP Palette` any file not having exactly this string at the first line will not be accepted as `.gpl` file
+ 2. A list of attributes separated by `:`.
+ 3. An optional comment line as separator.
+ 4. The colors of the palette as space separated `RGB` values together with a name
+
+The parser checks each line apart from the first line in the following order:
+
+ 1. Can line be parsed as **comment** (i.e. first character is `#`)
+ 2. Can line be parsed as **attribute** (i.e. two string separated by `:`)
+ 3. Can line be parsed as **color** (i.e. `RRR GGG BBB Name`)
+ 4. Otherwise abort parsing
+
+In case Xournal++ cannot parse the palette file (i.e. it cannot parse a line), it loads the default palette.
+If this happens there are two ways to get your palette file to load.
+
+Option 1, Debugging:
+
+ - run `xournalpp` from command line 
+ - check the error message (it tells you which line was not parsable)
+ - fix line in your `palette.gpl`
+ - retry
+
+Option 2, Fresh Start:
+
+ - remove the `palette.gpl` file
+ - start `xournalpp` (this will create a new default `palette.gpl` file)
+ - adjust `palett.gpl` to your needs
+
+!!! note
+    
+    It's recommended to have at least 11 colors in your palette as the default toolbar has 11 ColorItems.
+    In case your palette has less, Xournal++ will show a warning in the CLI and just cycle through the palette.
```

Comment:
```suggestion
    In case your palette has fewer colors than the toolbar has ColorItems, Xournal++ will show a warning in the CLI and just cycle through the palette.
```